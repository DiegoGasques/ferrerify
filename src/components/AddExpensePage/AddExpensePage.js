import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "../ExpenseForm/ExpenseForm";
import { addExpense } from "../../store/actions/expenses.actions";

export const AddExpensePage = ({ addExpense, history }) => {
  const onSubmit = expense => {
    addExpense(expense);
    history.push("/");
  };

  return (
    <div className="AddExpensePage">
      <h1>Add Expense</h1>
      <ExpenseForm onSubmit={onSubmit} />
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addExpense: expense => dispatch(addExpense(expense))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
