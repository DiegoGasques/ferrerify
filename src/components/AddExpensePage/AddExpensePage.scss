.AddExpensePage {
}

.AddExpensePage__header {
  background: darken($secondary, 20%);
  padding: $xs-size $m-size;
  margin: $m-size 0;
  -webkit-box-shadow: $main-shadow;
  box-shadow: $main-shadow;

  h2 {
    color: $snow;
    font-size: $h3;
  }
}
