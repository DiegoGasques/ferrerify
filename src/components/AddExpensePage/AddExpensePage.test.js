import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { AddExpensePage } from "./AddExpensePage";

describe("[COMPONENT] AddExpensePage", () => {
  it("Tests for correct render", () => {
    const wrapper = shallow(
      <AddExpensePage addExpense={() => {}} history={{ push: () => {} }} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for correct onSubmit call", () => {
    const expense = {
      description: "New Description",
      note: "New Note",
      amount: "67368",
      createdAt: 1283773
    };
    const addExpenseSpy = jest.fn();
    const historyPushSpy = jest.fn();
    const wrapper = shallow(
      <AddExpensePage
        addExpense={addExpenseSpy}
        history={{ push: historyPushSpy }}
      />
    );

    wrapper.find("ExpenseForm").prop("onSubmit")(expense);
    expect(addExpenseSpy).toHaveBeenCalledWith(expense);
    expect(historyPushSpy).toHaveBeenCalledWith("/");
  });
});
