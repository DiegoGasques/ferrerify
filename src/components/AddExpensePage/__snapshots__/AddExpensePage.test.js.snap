// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] AddExpensePage Tests for correct render 1`] = `
<div
  className="AddExpensePage"
>
  <Container
    column={true}
  >
    <div
      className="AddExpensePage__header"
    >
      <h2>
        Add Expense
      </h2>
    </div>
    <ExpenseForm
      onSubmit={[Function]}
    />
  </Container>
</div>
`;
